src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -224,6 +225,19 @@ def get_location_type_with_default_from_resource_group(cli_ctx):
     options_list=['--object-id', '-i'],
     help='The unique ID of the Azure AD administrator.')
 
+read_scale_param_type = CLIArgumentType(
+    options_list=['--read-scale'],
+    help='If enabled, connections that have application intent set to readonly '
+    'in their connection string may be routed to a readonly secondary replica. '
+    'This property is only settable for Premium and Business Critical databases.',
+    arg_type=get_three_state_flag('Enabled','Disabled', False, True))
```

Comment:
The right way to use get_three_state_flag is  arg_type=get_three_state_flag()

Reply:
I'm not following this comment.  I looked up the definition of get_three_state_flag()--pasted below--and this is exactly how it is defined.  In fact, the example given is 'enabled' and 'disabled.'  I have tested it as well and it is working as expected.  Can you please clarify what your request is?

def get_three_state_flag(positive_label='true', negative_label='false', invert=False, return_label=False):
    """ Creates a flag-like argument that can also accept positive/negative values. This allows
    consistency between create commands that typically use flags and update commands that require
    positive/negative values without introducing breaking changes. Flag-like behavior always
    implies the affirmative unless invert=True then invert the logic.
    - positive_label: label for the positive value (ex: 'enabled')
    - negative_label: label for the negative value (ex: 'disabled')
    - invert: invert the boolean logic for the flag
    - return_label: if true, return the corresponding label. Otherwise, return a boolean value

Reply:
You can use Enabled and Disabled, but I suggest the default true and false, which is consistent with other boolean parameters.
Maybe you should use Enum type if you expect a value of ['Enabled', 'Disabled'].

Reply:
It's not a boolean parameter, it's an enum all the way from API through SDK and all clients.  I originally had it as an enum and was asked to change to use 3 value type.  Are you suggesting it needs to go back to enum?

Reply:
Oh, I am a rookie. I haven't read the previous conversation. You can listen to experienced reviewers like juliehzl

Reply:
I'm actually in agreement with you.  I don't know that it was necessary to change from enum to three_state_flag, but I wasn't against doing it.  @Juliehzl can you please reply or approve?  It's been a week since the last comment.

Reply:
Hi @bradrich-msft , I know it is not boolean. But the functionality of this parameter is the same as other boolean parameter and as what @qwordy said above, using true and false will be consistent with others. You can refer to the parameter `--enable-llarg-file-share` in  `az storage account create`  and the implementation [here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/storage/operations/account.py#L78).
In addition, you should use get_enum_type(['Enabled', 'Disabled']) if you still want to use enum values.

Reply:
I've changed to enum type for consistency across our API, SDK, and other clients.